From: Matthias Bach <marix@marix.org>
Date: 2017-11-26 20:45:00 +0100
Subject: Fix compile on kernel 4.14
Upstream: to be done

The module no longer compiled on 4.14

diff -Naru hdjmod-1.28o/bulk.c hdjmod-1.28/bulk.c
--- hdjmod-1.28o/bulk.c	2009-01-27 15:25:50.000000000 +0100
+++ hdjmod-1.28/bulk.c	2017-11-26 20:41:31.941557776 +0100
@@ -23,6 +23,7 @@
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
+#include <linux/sched.h>
 #include <linux/slab.h>
 #include <linux/module.h>
 #include <linux/kref.h>
diff -Naru hdjmod-1.28o/device.c hdjmod-1.28/device.c
--- hdjmod-1.28o/device.c	2009-01-27 15:25:50.000000000 +0100
+++ hdjmod-1.28/device.c	2017-11-26 20:38:50.262807031 +0100
@@ -29,6 +29,7 @@
 #include <linux/version.h>	/* For LINUX_VERSION_CODE */
 #include <linux/errno.h>
 #include <linux/init.h>
+#include <linux/sched.h>
 #include <linux/slab.h>
 #include <linux/module.h>
 #include <linux/kref.h>
diff -Naru hdjmod-1.28o/midicapture.c hdjmod-1.28/midicapture.c
--- hdjmod-1.28o/midicapture.c	2009-01-27 15:25:50.000000000 +0100
+++ hdjmod-1.28/midicapture.c	2017-11-26 20:42:47.007150108 +0100
@@ -27,6 +27,7 @@
 #include <linux/version.h>	/* For LINUX_VERSION_CODE */
 #include <linux/errno.h>
 #include <linux/init.h>
+#include <linux/sched.h>
 #include <linux/slab.h>
 #include <linux/module.h>
 #include <linux/kref.h>
diff -Naru hdjmod-1.28o/midirender.c hdjmod-1.28/midirender.c
--- hdjmod-1.28o/midirender.c	2009-01-27 15:25:50.000000000 +0100
+++ hdjmod-1.28/midirender.c	2017-11-26 20:43:18.906131711 +0100
@@ -26,6 +26,7 @@
 #include <linux/version.h>	/* For LINUX_VERSION_CODE */
 #include <linux/errno.h>
 #include <linux/init.h>
+#include <linux/sched.h>
 #include <linux/slab.h>
 #include <linux/module.h>
 #include <linux/kref.h>
